packages/o11ytsdb/src/column-store.ts

```
@@ -0,0 +1,432 @@
+          }
+        }
+      }
+    } else {
+      // ── Path B: Individual decode (batch decode amortized by caller if needed) ──
+      for (const chunk of s.frozen) {
+        const tsChunk = group.frozenTimestamps[chunk.tsChunkIndex]!;
+        if (tsChunk.maxT < start || tsChunk.minT > end) continue;
+
+        // Decompress timestamps if needed.
+        const timestamps = tsChunk.timestamps
+          ?? (tsChunk.timestamps = this.tsCodec!.decodeTimestamps(tsChunk.compressed!));
```

Biome (2.4.11)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)